pkg/services/sqlstore/migrations/playlist_mig.go

```diff
@@ -43,4 +43,33 @@ func addPlaylistMigrations(mg *Migrator) {
 		{Name: "value", Type: DB_Text, Nullable: false},
 		{Name: "title", Type: DB_Text, Nullable: false},
 	}))
+
+	// Replacing auto-incremented playlistIDs with string UIDs
+	mg.AddMigration("Add UID column to playlist", NewAddColumnMigration(playlistV2, &Column{
+		Name: "uid", Type: DB_NVarchar, Length: 80, Nullable: false,
+	}))
+
+	// copy the (string representation of) existing IDs into the new uid column.
+	mg.AddMigration("Update uid column values in playlist", NewRawSQLMigration("").
+		SQLite("UPDATE playlist SET uid=printf('%09d',id) WHERE uid IS NULL;").
+		Postgres("UPDATE playlist SET uid=lpad('' || id::text,9,'0') WHERE uid IS NULL;").
+		Mysql("UPDATE playlist SET uid=lpad(id,9,'0') WHERE uid IS NULL;"))
+
+	mg.AddMigration("Add index for uid in playlist", NewAddIndexMigration(playlistV2, &Index{
+		Cols: []string{"uid"}, Type: IndexType,
```

Comment:
I think that we want a unique index type here to make prevent any dublications.
```suggestion
		Cols: []string{"uid"}, Type: UniqueIndex,
```

Also for other components [we require that these identifiers are unique per organisation](https://github.com/grafana/grafana/blob/49883501605177bf1e9336b066d02ebdcd56bcb9/pkg/services/sqlstore/migrations/datasource_mig.go#L152) but I'm not sure that we want to do the same for the newly introduced ones.